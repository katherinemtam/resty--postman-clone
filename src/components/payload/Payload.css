.Payload {
  border:#D3D3D3 1px solid;
  background-color: white;
  margin: 2em;
  padding: 1em;
  overflow: auto;

  width: 80vh;
  height: 50%;
}
