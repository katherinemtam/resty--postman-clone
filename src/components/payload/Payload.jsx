import React from 'react';
import PropTypes from 'prop-types';
import ReactJson from 'react-json-view';
import styles from './Payload.css';

const Payload = ({ payload }) =>  (
  <>
    <div data-testid="payload" className={styles.Payload}>
      <ReactJson src={payload} displayDataTypes={false} />
    </div>
  </>
);

Payload.propTypes = {
  payload: PropTypes.object
};

export default Payload;
