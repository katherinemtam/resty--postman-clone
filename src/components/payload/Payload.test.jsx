import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Payload from './Payload';

describe('Payload component', () => {
  afterEach(() => cleanup());
  it('renders Payload', () => {
    const { asFragment } = render(<Payload />);
    expect(asFragment()).toMatchSnapshot();
  });
});
