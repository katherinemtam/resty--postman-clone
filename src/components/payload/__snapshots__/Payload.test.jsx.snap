// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payload component renders Payload 1`] = `
<DocumentFragment>
  <div
    data-testid="payload"
  />
</DocumentFragment>
`;
